### PR TITLE
feat(service): add lark service

### PR DESCRIPTION
**Pull request by immanuelhume**
## Description

Adds a notifier service for [Lark](https://www.larksuite.com/).

## Motivation and Context

This addresses https://github.com/nikoksr/notify/issues/268.

Lark has three kinds of bots, referred to as "webhook", "custom app", and "store app" bots. The first two are supported through this PR, which explains the two different files `webhook.go` and `custom_app.go`. The store apps have a tricky API which would work against the simple style of this library so they are not included here.

## How Has This Been Tested?

Basic unit tests using mocks have been written. Manual testing has also been performed.

## Screenshots / Output (if appropriate):
The first screenshot is with a webhook. The second is with a custom app.

photo_2022-06-05_23-38-53
photo_2022-06-05_23-39-51

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: service/lark/common.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// https://open.larksuite.com/document/home/user-identity-introduction/introduction.
+type receiverIDType string
+
+const (
+	openID  receiverIDType = "open_id"
+	userID  receiverIDType = "user_id"
+	unionID receiverIDType = "union_id"
+	email   receiverIDType = "email"
+	chatID  receiverIDType = "chat_id"
+)
```
Comment (nikoksr) [inline service/lark/common.go:58]: I added the receiverIDType to all constants due to [this](https://staticcheck.io/docs/checks#SA9004). Could you please verify that this does not break functionality and/or behavior?

**Comment by nikoksr** [comment]
@immanuelhume so sorry for the very late response. 

I fixed the compatibility issues with the main branch and some linter warnings. Please check those before you push next time, please. 

I greatly appreciate your patience and contribution!

File: service/lark/common.go (hunk 1 above)
Comment (immanuelhume) [reply service/lark/common.go:58]: Tested it again manually and still works fine!

**Comment by immanuelhume** [comment]
@nikoksr Sorry I must've missed those warnings. Thanks for maintaining this project!

**Pull request merged**